### PR TITLE
ansible-test: remove Ubuntu 20.04 LTS

### LINKED ISSUE #80853: Remove Ubuntu 20.04 VM from ansible-test
### Summary

Remove Ubuntu 20.04 VM support from ansible-test.

It is only used for controller-side testing and does not provide Python 3.10.

### Issue Type

Feature Idea

### Component Name

`ansible-test`

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Fixes: #80853

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
.azure-pipelines/azure-pipelines.yml
changelogs/fragments/ansible-test-remove-ubuntu-2004.yml
test/integration/targets/apt/vars/Ubuntu-20.yml
test/lib/ansible_test/_data/completion/docker.txt

### CODE REVIEW

File: .azure-pipelines/azure-pipelines.yml [hunk 1]
```diff
@@ -149,8 +147,6 @@ stages:
               test: fedora37
             - name: openSUSE 15
               test: opensuse15
-            - name: Ubuntu 20.04
-              test: ubuntu2004
```
Comment (mattclay) [inline .azure-pipelines/azure-pipelines.yml]: These lines should not be removed. We only want to remove the Ubuntu 20.04 VM, not containers.

File: changelogs/fragments/ansible-test-remove-ubuntu-2004.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - ansible-test - Removed Ubuntu 20.04 LTS image from the default completion list.
```
Comment (mattclay) [inline changelogs/fragments/ansible-test-remove-ubuntu-2004.yml]:
```suggestion
  - ansible-test - Removed Ubuntu 20.04 LTS image from the `--remote` option.
```

File: test/integration/targets/apt/vars/Ubuntu-20.yml [hunk 3]
```diff
@@ -1 +0,0 @@
-multiarch_test_pkg: libunistring2
```
Comment (mattclay) [inline test/integration/targets/apt/vars/Ubuntu-20.yml]: This file is still needed for integration tests run in a container.

File: test/lib/ansible_test/_data/completion/docker.txt [hunk 4]
```diff
@@ -5,5 +5,4 @@ alpine3 image=quay.io/ansible/alpine3-test-container:5.0.0 python=3.10 cgroup=no
 centos7 image=quay.io/ansible/centos7-test-container:5.0.0 python=2.7 cgroup=v1-only
 fedora37 image=quay.io/ansible/fedora37-test-container:5.0.0 python=3.11
 opensuse15 image=quay.io/ansible/opensuse15-test-container:6.0.0 python=3.6
-ubuntu2004 image=quay.io/ansible/ubuntu2004-test-container:5.0.0 python=3.8
```
Comment (mattclay) [inline test/lib/ansible_test/_data/completion/docker.txt]: Container support should be retained. The VM completion entry should be removed instead:

https://github.com/ansible/ansible/blob/cde15f3c8158467a96023a9cffcba4bc0a207b0f/test/lib/ansible_test/_data/completion/remote.txt#L17

**Review by mattclay — CHANGES_REQUESTED** [review]

**Review by mattclay — APPROVED** [review]
Looks good.

CC @felixfontein I noticed you're using this here: https://github.com/ansible-collections/community.crypto/blob/230f0b51f2b4003089beb6767d7c0f7e5348a9cc/.azure-pipelines/azure-pipelines.yml#L205-L206

**Comment by felixfontein** [comment]
@mattclay thanks for the heads up! I just removed it in https://github.com/ansible-collections/community.crypto/pull/623.

**Pull request merged**